src/Data/Managers/V2/POST/UpdateCatalog/useUpdateCatalog.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  UseMutationResult,
+  useMutation,
+  useQueryClient,
+} from "@tanstack/react-query";
+import { PrimaryBaseUrlManager } from "@/UI";
+import { useFetchHelpers } from "../../helpers";
+
+/**
+ * React Query hook for updating environment catalog.
+ *
+ * @param {string} environment  - The environment to use for creating headers.
+ * @returns {UseMutationResult<void, Error, void, unknown>}- The mutation object from `useMutation` hook.
+ */
+export const useUpdateCatalog = (
```

Comment:
The name might be a bit confusing, perhaps we could go for something closer to the endpoint name, like useExportCatalog. I'd avoid the update keyword since it's a post call.